index.js

```diff
@@ -1,2 +1,544 @@
+ *   which to calculate the counter value. Defaults to 0 (no offset).
+ * @param {Integer} [options.counter] Counter value, calculated by default.
+ * @param {Integer} [options.digits=6] The number of digits for the one-time
+ *   passcode.
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @param {String} options.key (DEPRECATED. Use `secret` instead.)
+ *   Shared secret key
+ * @param {Integer} [options.initial_time=0] (DEPRECATED. Use `epoch` instead.)
+ *   Initial time since the UNIX epoch from which to calculate the counter
+ *   value. Defaults to 0 (no offset).
+ * @param {Integer} [options.length=6] (DEPRECATED. Use `digits` instead.) The
+ *   number of digits for the one-time passcode.
+ * @return {String} The one-time passcode.
+ */
+
+exports.totp = function totpGenerate (options) {
+
+  // shadow options
+  options = Object.create(options);
+
+  // calculate default counter value
+  if (options.counter == null) options.counter = exports._counter(options);
+
+  // pass to hotp
+  return this.hotp(options);
+};
+
+/**
+ * Verify a time-based One Time passcode and return the delta.
+ *
+ * @param {Object} options
+ * @param {String} options.secret Shared secret key
+ * @param {String} options.token Passcode to validate
+ * @param {Integer} [options.time] Time with which to calculate counter value.
+ *   Defaults to `Date.now()`.
+ * @param {Integer} [options.step=30] Time step in seconds
+ * @param {Integer} [options.epoch=0] Initial time since the UNIX epoch from
+ *   which to calculate the counter value. Defaults to 0 (no offset).
+ * @param {Integer} [options.counter] Counter value, calculated by default.
+ * @param {Integer} [options.digits=6] The number of digits for the one-time
+ *   passcode.
+ * @param {Integer} [options.window=0] The allowable margin for the counter.
+ *   The function will check "W" codes in the future and the past against the
+ *   provided passcode, e.g. if W = 5, and C = 1000, this function will check
+ *   the passcode against all One Time Passcodes between 995 and 1005,
+ *   inclusive.
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @return {Object} On success, returns an object with the time step
+ *   difference between the client and the server as the `delta` property.
+ * @method totp․verifyDelta
+ * @global
+ */
+
+exports.totp.verifyDelta = function totpVerifyDelta (options) {
+
+  // shadow options
+  options = Object.create(options);
+
+  // unpack options
+  var window = parseInt(options.window || 0, 10);
+
+  // calculate default counter value
+  if (options.counter == null) options.counter = exports._counter(options);
+
+  // adjust for two-sided window
+  options.counter -= window;
```

Comment:
Yep, this adjustment is to allow reuse of the `hotp.verifyDelta()` function.

Before range: [counter, counter+window]
After adjustment: [counter-window, counter+window]